From 6c0ded7c5a127923fc174640a02ad1c3c98fc994 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Mon, 13 Sep 2010 12:42:29 +0000
Subject: [PATCH] Makefile: only add gettext tests on XGETTEXT_INCLUDE_TESTS=YesPlease
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the "pot" target so that the xgettext(1) invocation no longer
include test files from t/t0200.

These should only be included in the Icelandic is.po file since Git's
own test suite uses them for regression tests, but nobody else should
translate them. Previously I indicated this with a TRANSLATORS comment
for all the strings:

    #. TRANSLATORS: This is a test. You don't need to translate it.
    #, c-format
    msgid "TEST: A C test string %s"
    msgstr ""

But if translators follow those instructions they'll never have 100%
translation coverage. It's much better to add a hidden option so that
is.po now has to use a special git.pot file for msgmerge(1):

    make pot XGETTEXT_INCLUDE_TESTS=YesPlease

But everything else can continue to use the documented:

    make pot

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

On Mon, Sep 13, 2010 at 07:15, Peter Krefting <peter@softwolves.pp.se> wrote:
> Ævar Arnfjörð Bjarmason:
>> * You're translating the TEST: messages, it's redundant except for
>>  Icelandic (which uses it for git.git tests), but I can see how it
>>  can be painful to have non-100% translation coverage.
>
> I just copied them verbatim, I didn't actually translate them. Without
> translating them I can't tell if I've translated everything when doing a
> "msgfmt -vvv --check sv.po".

This is a much better solution then. I'll submit this as part of an
updated ab/i18n series.

 Makefile |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 8cec626..680e578 100644
--- a/Makefile
+++ b/Makefile
@@ -2047,9 +2047,15 @@ XGETTEXT_OPTIONS_C = $(XGETTEXT_OPTIONS) --keyword=_ --keyword=N_ --language=C
 XGETTEXT_OPTIONS_SH = $(XGETTEXT_OPTIONS) --language=Shell
 XGETTEXT_OPTIONS_PERL = $(XGETTEXT_OPTIONS) --keyword=__ --language=Perl
 
-LOCALIZED_C = $(C_OBJ:o=c) t/t0200/test.c
-LOCALIZED_SH = $(SCRIPT_SH) t/t0200/test.sh
-LOCALIZED_PERL = $(SCRIPT_PERL) t/t0200/test.perl
+LOCALIZED_C = $(C_OBJ:o=c)
+LOCALIZED_SH = $(SCRIPT_SH)
+LOCALIZED_PERL = $(SCRIPT_PERL)
+
+ifdef XGETTEXT_INCLUDE_TESTS
+LOCALIZED_C += t/t0200/test.c
+LOCALIZED_SH += t/t0200/test.sh
+LOCALIZED_PERL += t/t0200/test.perl
+endif
 
 pot:
 	$(XGETTEXT) $(XGETTEXT_OPTIONS_C) $(LOCALIZED_C)
-- 
1.7.2.3.313.gcd15

